alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -141,13 +143,38 @@ const TeamManager& RunnableObject::getTeamManager() const
     return *_teamManager;
 }
 
+void TraceRunnableObject::traceException(const std::string& exceptionOriginMethod, const std::string& details)
+{
+    std::string msg = "Exception thrown from: " + getName() + "'s " + exceptionOriginMethod + " method";
+    if (!details.empty()) {
+        msg.append(", details: " + std::move(details));
+    }
+    Logging::logFatal(LOGNAME) << msg;
+}
+
+void RunnableObject::handleException(const std::string& exceptionOriginMethod, std::exception_ptr eptr)
+{
+    std::string details;
+    try {
+        std::rethrow_exception(eptr);
+    } catch (const std::exception& e) {
+        details = e.what();
+    } catch (...) {
+    }
+    _runnableObjectTracer.traceException(exceptionOriginMethod, details);
+    _runnableObjectTracer.finishTrace();
+    std::rethrow_exception(eptr);
```

Comment:
rethrowing inside this function obscures it a bit as opposed to having this function focus on the tracing and then explicitly rethrowing outside

Reply:
the `traceException` method is exclusively meant for tracing the exception. Here the idea is to basically handle the exception & if it can't be handled then I think we should throw here instead of at the caller site.